### DISCUSSION TITLE
How does vkvg compares with something like GTK or QT?

### DISCUSSION THREAD

**Discussion by rempas** [jpbruyere/vkvg#136] [Q&A] [answered]
Hello! I'm not sure that I understand but how does vkvg compares with something like the mentioned libraries? Are they the same thing?

**Comment by jpbruyere** [comment, marked as answer]
Gtk and QT are widget toolkits, they both have a 2d drawing libraries to draw graphic components. Gtk use cairo and QT has its own. Vkvg, as cairo or skia, is only meant to draw 2d shapes and text.

**Reply by rempas** [reply]
Oh, interesting! So, if I understand it right, Vkvg cannot be used standalone to create full applications and something else needs to be used. For example, something that uses that Xlib, XCB or Wayland (on Linux) to create and manage windows and then use Vkvg to draw things in the screen. Or am I mistaken?

**Reply by jpbruyere** [reply]
You need to create a valid vulkan context, either a window surface or a texture. To draw on a window, I use GLFW library that gives you also input devices support (keyboard, mouse,...). If you only want to draw on a texture and save the result on disk, you may go only with vkvg.

**Reply by rempas** [reply]
Alright, I got it! Thank you!

**Comment by houstudio** [comment]
I want to use VKVG in my CDroid project. Is it possible that VKVG supports a 3x3 matrix like Qt/Skia/Blend2D?

**Reply by jpbruyere** [reply]
You need the projective component in the matrix? For now, the rendering pipeline use only the 3x2 affine matrix as in Cairo. Implementing a 3x3 would require non trivial work. Without the need of project transform, you may easily translate the 3x3 into a 3x2 (row-maj).
cdroid looks an amazing project, but for now, be aware that I'm not very active on vkvg (would need to find funding if I want to), and vkvg is maybe not mature enough (vkvg is in alpha state) if you have some solutions in production.

**Reply by houstudio** [reply]
I’ll keep an eye on VKVG and Blend2D：）
